source/MaterialXRuntime/RtCommandEngine.h

```diff
@@ -0,0 +1,159 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_RTCOMMAND_H
+#define MATERIALX_RTCOMMAND_H
+
+/// @file
+/// Classes related to command execution.
+
+#include <MaterialXRuntime/RtPath.h>
+#include <MaterialXRuntime/RtPointer.h>
+
+namespace MaterialX
+{
+
+/// @class RtCommandResult
+/// Class holding the results of a command execution.
+class RtCommandResult
+{
+public:
+    /// Empty constructor.
+    RtCommandResult() :
+        _success(false)
+    {
+    }
+
+    /// Construct a command result from a boolean state and an optional error message.
+    RtCommandResult(bool success, const string& message = "") :
+        _success(success),
+        _message(message)
+    {
+    }
+
+    /// Construct a command result from a return object and optionally
+    /// a boolean state and an error message.
+    RtCommandResult(const RtObject& object, bool success = true, const string& message = "") :
+        _success(success),
+        _message(message),
+        _object(object)
+    {
+    }
+
+    /// Return true if the command executed succesfully.
+    bool success() const
+    {
+        return _success;
+    }
+
+    /// Return true if the command executed succesfully.
+    explicit operator bool() const
+    {
+        return success();
+    }
+
+    /// Return an error message if set when executing the command.
+    const string& getMessage() const
+    {
+        return _message;
+    }
+
+    /// Return an object resulting from executing the command.
+    const RtObject& getObject() const
+    {
+        return _object;
+    }
+
+private:
+    bool _success;
+    string _message;
```

Comment:
Can we change this to be a string vec. I've found cases where I wanted to collect multi-line errors such as during file read.

Reply:
We could but maybe wait until we have a specific use-case for that. Operations like file IO will not be done through commands since it should not be undoable.